joblib/hashing.py

```diff
@@ -194,7 +194,10 @@ def save(self, obj):
                 # https://github.com/numpy/numpy/issues/4983.  The
                 # workaround is to view the array as bytes before
                 # taking the memoryview
-                obj_bytes_view = obj.view(self.np.uint8)
+                if obj.flags.f_contiguous:
+                    obj_bytes_view = obj.T.view(self.np.uint8).T
+                else:
+                    obj_bytes_view = obj.view(self.np.uint8)
```

Comment:
It looks like we can simplify the code by instead of a try except do something like:

```python

if c_order:
    ...
elif fortran_order:
    ...
else:  # non contiguous array
    ...
```

Maybe there are some edge cases though, not entirely sure.